code/ai/aicode.cpp

```diff
@@ -14482,6 +14485,9 @@ void validate_mode_submode(ai_info *aip)
  */
 void ai_frame(int objnum)
 {
+	// Cyborg - can you believe this was added in *2022*??
+	Assertion (Objects[objnum].type == OBJ_SHIP, "Ai_frame was called for a non-ship of type %d. Please report!", Objects[objnum].type);
```

Comment:
Return here as well. We're about to index into `Ships` with this thing, if it isn't a ship for whatever reason we need to immediately return.

Reply:
100%.  Making that change now.